server/auth.js

```diff
@@ -1,137 +1,123 @@
-            (err) => {
-              if (err) {
-                return callback(Boom.unauthorized('Invalid token', 'Token'), null, null);
-              }
+            // this can throw if there is an error
+            await jwtVerifyAsync(token, resourceServerKey, jwtOptions.resourceServer.verifyOptions);
 
-              if (!decoded.payload.access_token || !decoded.payload.access_token.length) {
-                return callback(Boom.unauthorized('Invalid token', 'Token'), null, null);
-              }
+            if (decoded.payload.scope && typeof decoded.payload.scope === 'string') {
+              decoded.payload.scope = decoded.payload.scope.split(' '); // eslint-disable-line no-param-reassign
+            }
 
-              decoded.payload.scope = scopes.map(scope => scope.value);
-              return callback(null, true, decoded.payload);
+            return { credentials: decoded.payload, isValid: true };
+          }
+          if (isDashboardAdminRequest) {
+            if (!decoded.payload.access_token || !decoded.payload.access_token.length) {
+              return { isValid: false };
             }
-          );
+
+            // this can throw if there is an error
+            await jwtVerifyAsync(
+              token,
+              jwtOptions.dashboardAdmin.key,
+              jwtOptions.dashboardAdmin.verifyOptions
+            );
+
+            return { credentials: decoded.payload, isValid: true };
+          }
+        } catch (error) {
+          return { isValid: false };
         }
       }
-
-      return callback(null, false);
-    }
-  });
-  server.auth.default('jwt');
-  const session = {
-    register: tools.plugins.dashboardAdminSession,
-    options: {
-      stateKey: 'account-linking-admin-state',
-      sessionStorageKey: 'com.auth0.account_linking.admin_ui.session_token',
-      rta: config('AUTH0_RTA').replace('https://', ''),
-      domain: config('AUTH0_DOMAIN'),
-      scopes: '',
-      baseUrl: config('PUBLIC_WT_URL'),
-      audience: 'urn:api-account-linking',
-      secret: config('EXTENSION_SECRET'),
-      clientName: 'auth0-account-link',
-      onLoginSuccess: (decoded, req, callback) => {
-        if (decoded) {
-          decoded.scope = scopes.map(scope => scope.value); // eslint-disable-line no-param-reassign
-          return callback(null, true, decoded);
+    });
+    server.auth.default('jwt');
+
+    const session = {
+      plugin,
+      options: {
+        stateKey: 'account-linking-admin-state',
+        sessionStorageKey: 'com.auth0.account_linking.admin_ui.session_token',
+        rta: config('AUTH0_RTA').replace('https://', ''),
+        domain: config('AUTH0_DOMAIN'),
+        scopes: '',
+        baseUrl: config('PUBLIC_WT_URL'),
+        audience: 'urn:api-account-linking',
+        secret: config('EXTENSION_SECRET'),
+        clientName: 'auth0-account-link',
+        // eslint-disable-next-line no-unused-vars
+        onLoginSuccess: (decoded, req) => {
```

Comment:
nit: You should just be able to leave the second parameter `req` out and avoid the eslint override